backends/pipeline_api.py

```diff
@@ -30,7 +36,15 @@ def update_profile_from_edx(backend, user, response, is_new, *args, **kwargs):
 
     # this function is completely skipped if the backend is not edx or
     # the user has not created now
-    if backend.name != EdxOrgOAuth2.name or not is_new:
+    if backend.name != EdxOrgOAuth2.name:
+        return
+
+    if has_role(user, [Staff.ROLE_ID, Instructor.ROLE_ID]):
+        backend.strategy.session_set('next', '/learners')
+    else:
+        backend.strategy.session_set('next', '/dashboard')
+
+    if not is_new:
```

Comment:
`session_set` will set the next page after the user logs in via edX. The code should be executed regardless of whether `is_new` is true or not so I put it above the `is_new` check.